infra/src/main/scala/ch/epfl/scala/services/storage/sql/SqlRepo.scala

```diff
@@ -64,13 +65,22 @@ class SqlRepo(conf: DbConf) extends DatabaseApi {
     insertProject(NewProject.from(project))
 
   override def insertReleases(releases: Seq[NewRelease]): Future[Int] =
-    run(ReleaseTable.insertMany(releases))
+    releases
+      .grouped(SqlRepo.sizeOfInsertMany)
+      .map(r => run(ReleaseTable.insertMany(r)))
+      .sequence
+      .map(_.sum)
 
   def insertRelease(release: NewRelease): Future[NewRelease] =
     run(ReleaseTable.insert, release)
 
-  override def insertDependencies(deps: Seq[NewDependency]): Future[Int] =
-    run(DependenciesTable.insertMany(deps))
+  override def insertDependencies(deps: Seq[NewDependency]): Future[Int] = {
+    deps
+      .grouped(SqlRepo.sizeOfInsertMany)
+      .map(d => run(DependenciesTable.insertMany(d)))
+      .sequence
```

Comment:
IIUC, this is going to insert all the content of `deps` in parallel.

I guess your intention was to insert by batches of the size `sizeOfInsertMany`, instead? I don’t think the call to `grouped` helps here, because anyway the call to `map` won’t “wait” that the previous group finished being inserted before inserting the next one.

I think here the parallelism level (of the database operations) won’t be defined by `sizeOfInsertMany`, but by the size of the thread pool used by doobie in `run`.

Reply:
Maybe this is not important if that works, but I insist on the fact that `grouped(sizeOfInsertMany)` has no effect here. The calls to `run` will all happen concurrently.

Reply:
SQL allows to insert many rows at the same time using one connexion. 
in fact the size of the thread pool used by doobie controls parallelism , which 32. 
So we insert 10 000 rows in each connexion using a thread pool size of 32

Reply:
> Maybe this is not important if that works, but I insist on the fact that grouped(sizeOfInsertMany) has no effect here.

The `grouped(sizeOfInsertMany)` creates n request of 10000 entries. The n requests are submitted simultaneously. That's what we want.

Reply:
Ah, thank you for clarifying, I understand my mistake now! Indeed, there is a single SQL `INSERT` with 10k rows on every call to `run`, but all the calls to `run` happen concurrently (I thought we were inserting the rows one by one in the `insertMany` but this is not the case!).

Sorry for bringing confusion!